plugins/modules/identity/ipa/ipa_dnszone.py

```diff
@@ -117,7 +132,7 @@ def ensure(module, client):
         if not ipa_dnszone:
             changed = True
             if not module.check_mode:
-                client.dnszone_add(zone_name=zone_name, details={'idnsallowdynupdate': dynamicupdate})
+                client.dnszone_add(zone_name=zone_name, details={'idnsallowdynupdate': dynamicupdate, 'idnsallowsyncptr': allowsyncptr})
         else:
             changed = False
```

Comment:
In this case, I guess you have to compare the zone's `idnsallowsyncptr` state and update the zone if the state changes.

Reply:
Default behavior of CLI inherently handles this (given the value is provided).

Reply:
But here you are talking to the API directly, not to the CLI.

Reply:
Also, this code is only called when the zone does not exist. When the zone already exists, the module simply doesn't do anything and claims `changed=false`.

Reply:
Hi,

I wanted to touch base w/ the community on this PR.

I actually started digging into the code, and to @felixfontein 's point, the ipa_dnszone_find method only performs a broad, generic search returning whether zone exists or not (not as it relates to any parameters). If zone already exists (regardless of any associated parameters, it just sets changed to `false` as he notes above). The search should be expanded on whether the zone exists AND has requisite parameters set. 

Sidenote: This may also require changes in ipa.py to support the params dict for dnzone_find.

Reply:
@Akasurde @Nosmoht @fxfitz @justchris1 what do you think?

Reply:
@felixfontein  I made changes to include checking for generic zone existence, and a new function dnszone_mod() . Hopefully this helps start steering this in the right direction. 

Also, wasn't sure your thoughts on fixing dynamicupdate to also be a boolean (I'm not sure why it is a string?) but I didn't want to break backwards compatibility by making those changes, so I left that alone.

Reply:
> @Akasurde @Nosmoht @fxfitz @justchris1 what do you think?

I don't have any issue with the general direction it is going, but if we are going to do that it should be a comprehensive search.

Reply:
> Also, wasn't sure your thoughts on fixing dynamicupdate to also be a boolean (I'm not sure why it is a string?) but I didn't want to break backwards compatibility by making those changes, so I left that alone.

Since Ansible converts stings `true` and `false` to the associated booleans `True` and `False` when you specify `type='bool'`, this would be a backwards-compatible change. Asides from that, I think it is a very good idea to change this option's type to boolean.